utils/prng.go

```diff
@@ -1,73 +1,80 @@
 package utils
 
 import (
+	"crypto/rand"
 	"errors"
 	"golang.org/x/crypto/blake2b"
 )
 
 type PRNG interface {
-	GetClock() uint64
-	Seed(seed []byte)
-	GetSeed() []byte
 	Clock(sum []byte)
-	SetClock(sum []byte, n uint64) error
 }
 
-// prng is a structure storing the parameters used to securely and deterministically generate shared
+// OSrandPRNG is an empty struct used as a method receiver for crypto/rand Read
+type OSrandPRNG struct{}
+
+// Clock reads bytes from crypto/rand on sum.
+func (prng *OSrandPRNG) Clock(sum []byte) {
+	if _, err := rand.Read(sum); err != nil {
+		panic("crypto rand error")
+	}
+}
+
+// seededPRNG is a structure storing the parameters used to securely and deterministically generate shared
 // sequences of random bytes among different parties using the hash function blake2b. Backward sequence
 // security (given the digest i, compute the digest i-1) is ensured by default, however forward sequence
-// security (given the digest i, compute the digest i+1) is only ensured if the prng is keyed.
-type prng struct {
+// security (given the digest i, compute the digest i+1) is only ensured if the seededPRNG is keyed.
+type SeededPRNG struct {
```

Comment:
Does this type needs to be public if it is hidden behind the PRNG interface ?

Reply:
It needs to be exposed as it implements some functions outside the interface that might be called